### ISSUE TITLE
On newer versions of openmmtools, GPU can't be used if using mixed precision due to mixed precision check

### ISSUE THREAD

**Issue by aced125** [choderalab/openmmtools#505]
https://github.com/choderalab/openmmtools/blob/3f2cfc0d9ea5ee69b270b1f7705867548f8ca037/openmmtools/utils.py#L513

This function always fails due to this part:

```python
    if platform.getName() in ['CUDA', 'OpenCL']:
        from simtk import openmm
        properties = { 'Precision' : precision }
        system = openmm.System()
        integrator = openmm.VerletIntegrator(0.001)
        try:
            context = openmm.Context(system, integrator, properties)
            del context, integrator
            return True
        except Exception as e:
            return False
```

It always returns 

```
TypeError: Wrong number or type of arguments for overloaded function 'new_Context'.
  Possible C/C++ prototypes are:
    OpenMM::Context::Context(OpenMM::System const &,OpenMM::Integrator &)
    OpenMM::Context::Context(OpenMM::System const &,OpenMM::Integrator &,OpenMM::Platform &)
    OpenMM::Context::Context(OpenMM::System const &,OpenMM::Integrator &,OpenMM::Platform &,std::map< std::string,std::string,std::less< std::string >,std::allocator< std::pair< std::string const,std::string > > > const &)
    OpenMM::Context::Context(OpenMM::Context const &)
```

Try it:

```python
precision = "mixed"
from simtk import openmm
properties = { 'Precision' : precision }
system = openmm.System()
integrator = openmm.VerletIntegrator(0.001)
context = openmm.Context(system, integrator, properties)
```

Thus, whenever configuring the platform, if trying to check available platforms that allow mixed precision, the CUDA and OpenCL ones will always fail.

**Comment by jchodera** [comment]
Thanks for the fantastic bug report and minimal example to reproduce!

The syntax for [line 551](https://github.com/choderalab/openmmtools/blob/3f2cfc0d9ea5ee69b270b1f7705867548f8ca037/openmmtools/utils.py#L551) should be changed from
```python
context = openmm.Context(system, integrator, properties)
```
to
```python
context = openmm.Context(system, integrator, platform, properties)
```
I'll open a PR.

**Comment by aced125** [comment]
Hey @jchodera I tried this, I then get a new error, unfortunately. It then says:

```python
precision = "mixed"
from simtk import openmm

platform = openmm.Platform.getPlatformByName("CUDA")
properties = { 'Precision' : precision }
system = openmm.System()
integrator = openmm.VerletIntegrator(0.001)
context = openmm.Context(system, integrator, platform, properties)
```

```
Exception: Cannot create a Context for a System with no particles
```

When I try the following: 

```python
precision = "mixed"
from simtk import openmm
import openmmtools as mmt

platform = openmm.Platform.getPlatformByName("CUDA")
properties = { 'Precision' : precision }
system = mmt.testsystems.HarmonicOscillator()
integrator = openmm.VerletIntegrator(0.001)
context = openmm.Context(system, integrator, platform, properties)
```

I get: 

```
NotImplementedError: Wrong number or type of arguments for overloaded function 'new_Context'.
  Possible C/C++ prototypes are:
    OpenMM::Context::Context(OpenMM::System const &,OpenMM::Integrator &)
    OpenMM::Context::Context(OpenMM::System const &,OpenMM::Integrator &,OpenMM::Platform &)
    OpenMM::Context::Context(OpenMM::System const &,OpenMM::Integrator &,OpenMM::Platform &,std::map< std::string,std::string,std::less< std::string >,std::allocator< std::pair< std::string const,std::string > > > const &)
    OpenMM::Context::Context(OpenMM::Context const &)
```

**Comment by jchodera** [comment]
Now I'm super confused---even `context = openmm.Context(system, integrator)` does not work for me.

@peastman: Can you help us identify what is going on here? See the minimal examples above.

**Comment by peastman** [comment]
```python
system = mmt.testsystems.HarmonicOscillator()
```

HarmonicOscillator is not a System object.  You can get an actual System using its `system` field.

**Comment by jchodera** [comment]
@peastman : Of course, thank you! The swig error messages are not so helpful, so we easily confused two different failure mechanisms for being the same thing.

I've opened a PR to fix this in #506. Once we've verified everything is working, we can cut a new critical bugfix release.

**Comment by aced125** [comment]
Thanks @jchodera and @peastman for the very quick fix indeed!
